src/components/paginator-render/paginator-render.tsx

```diff
@@ -0,0 +1,668 @@
+import {
+  Component,
+  ComponentInterface,
+  Prop,
+  Event,
+  EventEmitter,
+  h,
+  Host,
+  State,
+  Watch
+} from "@stencil/core";
+import { tokenMap } from "../../common/utils";
+import {
+  COMBO_BOX_EXPORT_PARTS,
+  PAGINATOR_PARTS_DICTIONARY
+} from "../../common/reserved-names";
+import { PaginatorTranslations } from "./translations";
+import {
+  ACTUAL_PAGE,
+  AMOUNT_ITEMS_IN_PAGE,
+  TOTAL_ITEMS,
+  TOTAL_PAGES
+} from "./constants";
+import {
+  PaginatorRenderHyperlinkModel,
+  PaginatorRenderModel,
+  PaginatorRenderNumericModel,
+  ControlsOrder
+} from "./types";
+import { ComboBoxModel } from "../../components";
+
+@Component({
+  tag: "ch-paginator-render",
+  styleUrl: "paginator-render.scss",
+  shadow: true
+})
+export class ChPaginator implements ComponentInterface {
+  /**
+   * Specifies the index for the actual page in paginator
+   */
+  @State() actualPage: number = 1;
+
+  /**
+   * State to obtain the total pages on model,
+   * Use -1 if not known and 'hasNextPage' property to indicate
+   * that the end has been reached.
+   */
+  @State() totalPages: number = 1;
+
+  /**
+   * State to define the hyperlink or numeric model
+   */
+  @State() hasHyperlink: boolean = false;
+
+  /**
+   * Specifies the UI Model for the Paginator
+   */
+  @Prop()
+  readonly model: PaginatorRenderModel = [];
+
+  @Watch("model")
+  modelChanged() {
+    this.hasHyperlink = Array.isArray(this.model);
+
+    if (this.hasHyperlink) {
+      this.totalPages = (this.model as PaginatorRenderHyperlinkModel)?.length;
+    } else {
+      this.totalPages =
+        (this.model as PaginatorRenderNumericModel)?.totalPages ?? -1;
+    }
+  }
+
+  /**
+   * Specifies the Selected Page, can be a number or hyperlink, default number = 1.
+   */
+  @Prop({ mutable: true }) selectedPage: string | number = 1;
+
+  @Watch("selectedPage")
+  selectedPageChanged() {
+    if (this.hasHyperlink) {
+      this.actualPage =
+        (this.model as PaginatorRenderHyperlinkModel).indexOf(
+          this.selectedPage.toString()
+        ) + 1;
+    } else {
+      this.actualPage = Number(this.selectedPage);
+    }
+  }
+
+  /**
+   * Number for Total Items, default = undefined.
+   */
+  @Prop() readonly totalItems: number | undefined = undefined;
+
+  /**
+   * Number for Total Items Per Page, default = 10.
+   */
+  @Prop({ mutable: true }) itemsPerPage: number = 10;
+
+  /**
+   * Indicates that the end has been reached.
+   * Use when total pages are not known (totalPages = -1).
+   */
+  @Prop() readonly hasNextPage: boolean = false;
+
+  /**
+   * Number of pages to show starting from the selected page to the left, default = 3.
```

Comment:
We can avoid adding _"default = ..."_ in the description, since the default value is automatically generated as a separated column in the readmes.

![image](https://github.com/user-attachments/assets/29622843-d4ff-4011-9327-c1c20f8e0950)


```suggestion
   * Number of pages to show starting from the selected page to the left.
```

Reply:
done